source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -156,17 +155,18 @@ Http::FilterHeadersStatus FaultFilter::decodeHeaders(Http::RequestHeaderMap& hea
 
   absl::optional<std::chrono::milliseconds> duration = delayDuration(headers);
   if (duration.has_value()) {
-    delay_timer_ =
-        decoder_callbacks_->dispatcher().createTimer([this]() -> void { postDelayInjection(); });
+    delay_timer_ = decoder_callbacks_->dispatcher().createTimer(
+        [this, &headers]() -> void { postDelayInjection(headers); });
     ENVOY_LOG(debug, "fault: delaying request {}ms", duration.value().count());
     delay_timer_->enableTimer(duration.value(), &decoder_callbacks_->scope());
     recordDelaysInjectedStats();
     decoder_callbacks_->streamInfo().setResponseFlag(StreamInfo::ResponseFlag::DelayInjected);
     return Http::FilterHeadersStatus::StopIteration;
   }
 
-  if (isAbortEnabled()) {
-    abortWithHTTPStatus();
+  auto abort_code = abortHttpStatus(headers);
```

Comment:
nit: const auto. A few other cases of this elsewhere. Try to constify whatever you can.